File: java/numguess.java
import java.util.Scanner;
import java.util.Random;

public class numguess {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();
        boolean playAgain = true;

        System.out.println("Welcome to the Guess the Number game!");

        while (playAgain) {
            int attempts = 0;
            int targetNumber = random.nextInt(100) + 1;
            while (true) {
                System.out.print("Enter your guess (between 1 and 100): ");
                int userGuess = scanner.nextInt();
                attempts++;

                if (userGuess == targetNumber) {
                    System.out.println("Congratulations! You've guessed the number " + targetNumber + " correctly in " + attempts + " attempts!");
                    break;
                } else if (userGuess < targetNumber) {
                    System.out.println("Too low! Try guessing a higher number.");
                } else {
                    System.out.println("Too high! Try guessing a lower number.");
                }

                if (attempts == 10) {
                    System.out.println("Oops, you've reached the maximum number of attempts. The correct number was " + targetNumber + ".");
                    break;
                }
            }

            System.out.print("Do you want to play again? (yes/no): ");
            String playAgainInput = scanner.next();
            if (!playAgainInput.equalsIgnoreCase("yes")) {
                playAgain = false;
                System.out.println("Thanks for playing!");
            } else {
                System.out.println("\nStarting a new round...\n");
            }
        }
        scanner.close();
    }
}

File: C++/Numguess.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include <limits> 
using namespace std;

int main() {
    cout << "Welcome to the Guess the Number Game!" << endl;
    
    while (true) {
        srand(time(0)); 
        int lower_bound = 1;
        int upper_bound = 100;
        int attempts = 0;
        int target_number = lower_bound + rand() % (upper_bound - lower_bound + 1); // Generate random number
        
        cout << "I'm thinking of a number between " << lower_bound << " and " << upper_bound << ". Can you guess it?" << endl;
        
        while (true) {
            int user_guess;
            cout << "Enter your guess: ";
            cin >> user_guess;
            attempts++;
            
            if (cin.fail()) {
                cin.clear();
                cin.ignore(numeric_limits<streamsize>::max(), '\n');
                cout << "Invalid input. Please enter a valid number." << endl;
                continue;
            }
            
            if (user_guess < lower_bound || user_guess > upper_bound) {
                cout << "Please enter a number between " << lower_bound << " and " << upper_bound << "." << endl;
                continue;
            } else if (user_guess == target_number) {
                cout << "Congratulations! You guessed the number " << target_number << " in " << attempts << " attempts!" << endl;
                break;
            } else if (user_guess < target_number) {
                cout << "Too low! Try again." << endl;
            } else {
                cout << "Too high! Try again." << endl;
            }
        }
        
        char play_again;
        cout << "Do you want to play again? (yes/no): ";
        cin >> play_again;
        if (tolower(play_again) != 'y') {
            cout << "Thanks for playing!" << endl;
            break;
        }
    }
    
    return 0;
}
